checkov/common/util/secrets_omitter.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import annotations
+import logging
+from enum import Enum
+from typing import Iterator, TYPE_CHECKING
+
+from checkov.common.bridgecrew.check_type import CheckType
+
+if TYPE_CHECKING:
+    from checkov.common.output.record import Record
+    from checkov.common.output.report import Report
+
+
+class SecretsOmitterStatus(Enum):
+    SUCCESS = 0
+    INSUFFICIENT_REPORTS = 1
+
+
+class SecretsOmitter:
+    def __init__(self, reports: list[Report]):
+        self.reports: list[Report] = [report for report in reports if report.check_type != CheckType.SECRETS]
+        secrets_report = [report for report in reports if report.check_type == CheckType.SECRETS]
+        self.secrets_report: Report | None = secrets_report[0] if len(secrets_report) == 1 else None
+
+    def _secret_check(self) -> Iterator[Record]:
+        if not self.secrets_report:
+            # Should not reach here, used for typing
+            return
+
+        for check in self.secrets_report.failed_checks:
+            yield check
+
+    def _non_secret_check(self) -> Iterator[Record]:
+        for report in self.reports:
+            checks = report.failed_checks + report.passed_checks
+            for check in checks:
```

Comment:
```suggestion
            for check in itertools.chain(report.failed_checks, report.passed_checks):
```
no need to create a new list